### ISSUE TITLE
[ENH]:how to Color 3-D Bars by Height like matlab bar3 function?

### ISSUE THREAD

**Issue by renjf1** [matplotlib/matplotlib#23265]
### Problem

https://ww2.mathworks.cn/help/matlab/creating_plots/color-3-d-bars-by-height-1.html

how can i draw 3d bars gradient in matplotlib? the bar3d(color, cmap) is not working.

pls help.

**Comment by story645** [comment]
You should be able to pass in a list of colors as with the 2d case, so roughly `colors = camp(norm(heights))`

If that doesn't work, please post on discourse.matplotlib.org

That being said, I'm conflicted on natively supporting cmap for bar since it's unclear which value folks would want to colormap - how to decide if it's height, position, or a third value seperate from either . Though maybe it should be implemented like scatter where it's applied to data that's passed into the color keyword?

**Comment by renjf1** [comment]
@story645 thks for reply this thread.
test code:
```
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.colors import LinearSegmentedColormap
from matplotlib import style
style.use('ggplot')

# setup the figure and axes
fig = plt.figure(figsize=(8, 3))
ax1 = fig.add_subplot(111, projection='3d')

# fake data
_x = np.array([1,2,3,4])
_y = np.array([1,2,3,4,5])
_xx, _yy = np.meshgrid(_x, _y)
x, y = _xx.ravel(), _yy.ravel()

top = x + y
bottom = np.zeros_like(top)
width = depth = 1

norm = plt.Normalize(-1,1)
norm_values = norm(top)
map_vir = plt.cm.get_cmap(name='plasma')
colors = map_vir(norm_values)

ax1.bar3d(x, y, bottom, 0.5, 0.5, top,color=colors,shade=True)
plt.show()
```
I'm working on a project, like a car, it's has attributes as [speed, press,**statistics_value**] , the data value will be [20, 3000, 23.6]......   i want to draw a 3d bar to figure the datas. if **statistics_value <10 ,the color:'r', if statistics >50, the color:'dark red'**.  this is my purpose. 
Like scatter which give the data to color keyword is simple way, and that's what i want. but 4 lines code in matlab....

I will see the example code, but it's so long....

**Comment by story645** [comment]
Sorta like #22969, this should possibly be easier to implement now that the colorizer architecture exists.

**Comment by timhoffm** [comment]
I believe colormapped bars -  2d or 3d - are a niche topic. I wouldn’t invest time here right now. But won’t object if somebody capable can implement this without much guidance.

**Comment by story645** [comment]
> But won’t object if somebody capable can implement this without much guidance.

Yeah, I think the solution for #22969 could be a model here and this issue likely won't even require a new artist. I agree this is low priority, but I think it's in scope enough that a clean PR would be accepted.
